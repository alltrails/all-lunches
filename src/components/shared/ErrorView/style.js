import styled from 'styled-components';
import { colors, media, type } from 'styles';

export const ErrorViewWrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  position: absolute;
  height: 100vh;
  width: 100%;
  top: 0;
  left: 0;
  z-index: 4;
  padding: 20%;
  background: ${colors.base} center;
`;

export const ErrorText = styled.div`
  font-size: 1.6rem;
  font-weight: ${type.weight.semiBold};
  text-align: center;
  white-space: pre-wrap;

  ${media.min.baseline`font-size: 2rem;`}
`;
