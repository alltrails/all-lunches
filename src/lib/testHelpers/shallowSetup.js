import React from 'react';
import { shallow } from 'enzyme';

export default (Component, defaultProps) =>
  (otherProps = {}) => {
    const props = { ...defaultProps, ...otherProps };
    const wrapper = shallow(<Component {...props} />);

    return { props, wrapper };
  };
