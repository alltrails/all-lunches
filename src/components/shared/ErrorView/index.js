import React from 'react';
import PropTypes from 'prop-types';

import { ErrorText, ErrorViewWrapper } from './style';

const ErrorView = ({ text }) => (
  <ErrorViewWrapper>
    <ErrorText>
      {text ||
        'Looks like there was an issue. Please refresh the page or try again.'}
    </ErrorText>
  </ErrorViewWrapper>
);

ErrorView.propTypes = {
  text: PropTypes.string,
};

ErrorView.defaultProps = {
  text: null,
};

export default ErrorView;
