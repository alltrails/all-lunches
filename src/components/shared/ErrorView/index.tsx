import React from 'react';
import PropTypes from 'prop-types';

import { ErrorText, ErrorViewWrapper } from './style';

type ErrorViewProps = {
  text?: string;
};

const ErrorView = ({ text }: ErrorViewProps) => (
  <ErrorViewWrapper>
    <ErrorText>
      {text || 'Looks like there was an issue. Please refresh the page or try again.'}
    </ErrorText>
  </ErrorViewWrapper>
);

export default ErrorView;
