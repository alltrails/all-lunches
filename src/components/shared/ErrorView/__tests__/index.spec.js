/* eslint-env jest */

import shallowSetup from 'lib/testHelpers/shallowSetup';
import ErrorView from '../index';

describe('ErrorView', () => {
  const defaultProps = {};
  const setup = shallowSetup(ErrorView, defaultProps);

  it('renders the expected view when the text prop exists', () => {
    const { wrapper } = setup({ text: 'Sorry, an error occurred.' });
    expect(wrapper).toMatchSnapshot();
  });

  it('renders the expected view when the text prop does not exist', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });
});
