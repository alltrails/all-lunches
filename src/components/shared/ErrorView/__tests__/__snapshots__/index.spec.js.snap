// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorView renders the expected view when the text prop does not exist 1`] = `
<style__ErrorViewWrapper>
  <style__ErrorText>
    Looks like there was an issue. Please refresh the page or try again.
  </style__ErrorText>
</style__ErrorViewWrapper>
`;

exports[`ErrorView renders the expected view when the text prop exists 1`] = `
<style__ErrorViewWrapper>
  <style__ErrorText>
    Sorry, an error occurred.
  </style__ErrorText>
</style__ErrorViewWrapper>
`;
